yaksh/views.py

```diff
@@ -1320,3 +1341,48 @@ def design_questionpaper(request):
         context = {'form': form}
         return my_render_to_response('yaksh/design_questionpaper.html',
                                      context, context_instance=ci)
+
+@login_required
+def view_profile(request):
+    """ view moderators and users profile """
+
+    user = request.user
+    ci = RequestContext(request)
+    if not user.is_authenticated():
+        raise Http404('You are not allowed to view this page!')
+    else:
+        return my_render_to_response('yaksh/view_profile.html',
+                                    context_instance=ci)
+
+@login_required
+def edit_profile(request):
+    """ edit profile details facility for moderator and students """
+
+    context = {}
+    user = request.user
+    ci = RequestContext(request)
+    data = {}
+    if not user.is_authenticated():
+        raise Http404('You are not allowed to view this page!')
+    if request.method == 'POST':
+        form = EditProfile(request.POST)
+        if form.is_valid():
+            data = form.cleaned_data
+            form.save(user)
+            return my_render_to_response('yaksh/profile_updated.html',
+                                        context_instance=ci)
+        else:
+            context['form'] = form
+            return my_render_to_response('yaksh/editprofile.html', context,
+                                        context_instance=ci)
+    else:
+        data['first_name'] = user.first_name
+        data['last_name'] = user.last_name
+        data['institute'] = user.profile.institute
```

Comment:
Suppose in rare case if profile is not present, then user.profile will throw an exception.